docs/user/openstack/deploy_sriov_workers.md

```diff
@@ -0,0 +1,356 @@
+
+```yaml
+kind: MachineConfig
+apiVersion: machineconfiguration.openshift.io/v1
+metadata:
+  name: 99-vfio-noiommu
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  osImageURL: ''
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - filesystem: root
+        path: "/etc/modprobe.d/vfio-noiommu.conf"
+        contents:
+          source: data:text/plain;charset=utf-8;base64,b3B0aW9ucyB2ZmlvIGVuYWJsZV91bnNhZmVfbm9pb21tdV9tb2RlPTEK
+          verification: {}
+        mode: 0644
+```
+
+If you need to configure your deployment for real-time running or low latency, see the [performance addon operator documentation](perfocmance-addon-operator).
+
+## Creating SR-IOV Networks for Compute Machines
+
+The SR-IOV networks that are attached to compute nodes need to be created before creating the nodes. Please reference the example of how to create radio and uplink provider networks in OpenStack.
+
+```sh
+# Create Networks
+openstack network create radio --provider-physical-network radio --provider-network-type vlan --provider-segment 120
+openstack network create uplink --provider-physical-network uplink --provider-network-type vlan --external
+
+# Create Subnets
+openstack subnet create --network radio --subnet-range <radio_network_subnet_range> radio
+openstack subnet create --network uplink --subnet-range <uplink_network_subnet_range> uplink
+```
+
+## Creating SR-IOV Compute Nodes in IPI
+
+Creating compute nodes as a post-install operation in IPI is handled by the machine-api. To create a new set of compute nodes in the machine-api, you will need to [create a new machineSet in OpenShift](openstack-machine-sets). Note that it is much easier to copy the existing machineSet and edit the duplicate than to create one from scratch.
+
+```sh
+oc get machineset -n openshift-machine-api <machineset_name> -o yaml > sriov_machineset.yaml
+```
+
+When editing an existing machineSet, or a copy of one, to create SR-IOV compute nodes, add each subnet configured for SR-IOV to the machineSet's `providerSpec`. The example below will attach ports from the `radio` and `uplink` subnets, from the previous example, to all the compute nodes in the machineSet. The `nicType` needs to be set to `direct`, and the `portSecurity` needs to be set to `false` for all subnets that are using SR-IOV. The number of ports to attach for a given subnet or network is specified by setting `portCount`. For the sake of example, we attach four ports for each SR-IOV subnet on every machine. The tags added to the `portTags` list will be added to any port attached to machines from a given subnet or network. In this example, all ports attached to machines from the `radio` subnet will get the "sriov" and "radio" tags added to them, and all ports attached to machiens from the `uplink` subnet will get the "sriov" and "uplink" tags added to them. Lastly, be sure to set `trunk: false` since trunking is not supported with SR-IOV at this time.
```

Comment:
Another place where a list of actions is probably best. 
something like: 
the new sriov worker machineset should include the following in the providerSpec: 
- subnets to attach to the workers. 
  - nicType: direct
  - portSecurity: false
  - portCount: 
  - portTag
- trunk: false

Then explain in a sentence or two what it does.